### PR TITLE
UHF-8419: Header Top Menu nolink item padding fix

### PR DESCRIPTION
**Pull request by xkhaven**
# [UHF-8419](https://helsinkisolutionoffice.atlassian.net/browse/UHF-8419)

## What was done

* Added padding for <nolink> items on Header Top Menu on mobile layout

## How to install

* Make sure your instance is up and running on latest dev branch.
  * `git pull origin dev`
  * `make fresh`
* Update the HDBT theme
  * `composer require drupal/hdbt:dev-UHF-8419_Menu_nolink_padding`
* Run `make drush-cr`

## How to test

* [ ] Add new menu item to the Header Top Menu with <nolink> URL
* [ ] Check that new item's styles matches the other items in the menu on mobile layout
* [x] Check that code follows our standards

## Designers review

* [X] This PR does not need designers review

[UHF-8419]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-8419?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/scss/06_components/navigation/global/_mmenu.scss [hunk 1]
```diff
@@ -310,7 +310,8 @@ $-footerlink-target-size: 44px;
     padding: 0;
   }
 
-  .menu__link {
+  .menu__link,
+  span.menu__item--nolink {
```
[the remark this answers is not in the document]
Comment (xkhaven) [reply src/scss/06_components/navigation/global/_mmenu.scss:314]: This SPAN has parent with same CSS class ".menu__item--nolink" so styles would come as duplicate without targeting to HTML tag with same class.

[the remark this answers is not in the document]
Comment (xkhaven) [reply src/scss/06_components/navigation/global/_mmenu.scss]: Added new class to identify a child of this class selector.

File: src/scss/06_components/navigation/global/_mmenu.scss [hunk 2]
```diff
@@ -310,7 +310,8 @@ $-footerlink-target-size: 44px;
     padding: 0;
   }
 
-  .menu__link {
+  .menu__link,
+  .menu__nolink {
```
Comment (juho-lehmonen) [inline src/scss/06_components/navigation/global/_mmenu.scss:314]: Discussed this with Mikko, and the `menu__nolink` class should only have the padding and font-size styles, not any of the hover or focus styles (underline and outline).

**Review by juho-lehmonen — CHANGES_REQUESTED** [review]

File: src/scss/06_components/navigation/global/_mmenu.scss (hunk 2 above)
Comment (xkhaven) [reply src/scss/06_components/navigation/global/_mmenu.scss:314]: `.menu__nolink` is now as a own class.

**Review by juho-lehmonen — APPROVED** [review]

**Pull request merged**